app/src/main/java/com/example/bottomnav/CAN_Data.java

```diff
@@ -0,0 +1,48 @@
+package com.example.bottomnav;
+
+class CAN_Data {
+    int id;
+    int[] data;
+
+    public CAN_Data(int id, int[] data, int len) {
+        this.id = id;
+        this.data = data;
+    }
+
+    public static CAN_Data decode(String raw) {
+        int id = -1;
+        int len;
+
+        int index = -1;
```

Comment:
Can you leave this uninitialized? It doesn't really need to be -1?
Similarly with `id`.

Hopefully if the IDE is good, it can also warn you if you end up using an uninitialized variable, whereas if you give it a default it won't be able to warn you.